File: src/main/java/com/pancisin/bookster/models/enums/Unit.java
package com.pancisin.bookster.models.enums;

import com.fasterxml.jackson.annotation.JsonFormat;

@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum Unit {
	HOUR("HOUR", "unit.hour"), MINUTE("MINUTE", "unit.minute"), SQUARED_METER("SQUARED_METER",
			"unit.squared_meter"), PIECE("PIECE", "unit.piece");

	Unit(String name, String code) {
		this.name = name;
		this.code = code;
	}

	private final String name;
	private final String code;

	public String getName() {
		return name;
	}

	public String getCode() {
		return code;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/enums/Unit.kt
package com.pancisin.bookster.model.enums

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty

@JsonFormat(shape = JsonFormat.Shape.OBJECT)
enum class Unit private constructor(
  @JsonProperty("name")
  val prop: String,
  val code: String
) {
  HOUR("HOUR", "unit.hour"), MINUTE("MINUTE", "unit.minute"), SQUARED_METER("SQUARED_METER",
    "unit.squared_meter"),
  PIECE("PIECE", "unit.piece")
}
